lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1367,8 +1374,17 @@ result in the same scenario.
 sub carry_over_bugrefs {
     my ($self) = @_;
 
-    # the carry over makes only sense for some jobs
-    return if !grep { $_ eq $self->result } (FAILED, SOFTFAILED, INCOMPLETE);
+    # the carry over makes only sense for jobs with failed modules
+    # (but including ignore_failure fails, so can't check overall result)
+    my $gotfail = 0;
+    my $modules = $self->modules;
+    while (my $m = $modules->next) {
```

Comment:
Couldn't we check failure_reason to be 'GOOD' instead of duplicating the module loop? We have to calculcate it anyway - but the calculcation is hidden in carry_over_candidate.